src/main/java/com/fishercoder/solutions/_7.java

```diff
@@ -14,15 +14,22 @@
 
     public static class Solution1 {
         public int reverse(int x) {
-            long rev = 0;
-            while (x != 0) {
-                rev = rev * 10 + x % 10;
-                x /= 10;
-                if (rev > Integer.MAX_VALUE || rev < Integer.MIN_VALUE) {
-                    return 0;
-                }
+            long xL = x; // in case x=Integer.MIN_VALUE and cause error for 'x=-x' in line 7
```

Comment:
What's in line 7?

Reply:
We want to always handle a positive number and thus need to handle -x when x<0. But considering that -x will overflow when x=Integer.MIN_VALUE, we substitute a long type xL for int type x.